### ISSUE TITLE
"CREATE INDEX USING btree" failed

### ISSUE THREAD

**Issue by qinix** [pingcap/tidb#3363]
1. What did you do?
If possible, provide a recipe for reproducing the error.

```sql
CREATE INDEX `index_ali_logs_on_action_type` USING btree ON `ali_logs` (`action_type`);
```

2. What did you expect to see?

3. What did you see instead?

it says 
```
line 0 column 51 near " btree ON `ali_logs` (`action_type`)" (total length 87)
```

4. What version of TiDB are you using (`tidb-server -V`)?

```
Git Commit Hash: 64830c93b74356f2a6ab37a95a9c9f4113d1db7d
UTC Build Time:  2017-05-27 08:43:18
```

**Comment by holys** [comment]
@qinix  For above 4, please show specific version information by executing `tidb-server -V` like: 

```
Git Commit Hash: 2198d0e7521a5c6c493a4c7af58509b6b277fd6e
UTC Build Time:  2017-05-23 03:07:15
```

**Comment by qinix** [comment]
@holys 

```
Git Commit Hash: 64830c93b74356f2a6ab37a95a9c9f4113d1db7d
UTC Build Time:  2017-05-27 08:43:18
```

**Comment by shenli** [comment]
@qinix `USING btree` is meaningless in TiDB. We will support this in parser but ignore it.

**Comment by qinix** [comment]
@shenli 👍

**Comment by Li-Tao** [comment]
@shenli how we search index in tidb to get the qualified row key? As we know in traditional relational databases like mysql, the index leverages the btree fast searching mechanism to get the row key efficiently.

**Comment by jackysp** [comment]
In addition, `CREATE INDEX idx ON t (a) COMMENT 'foo'` is not supported, but `ALTER TABLE t ADD INDEX idx (a) USING BTREE COMMENT 'foo'` is supported #3803

**Comment by jackysp** [comment]
Fixed #3814
